examples/multiple_data_source_examples/public/components/data_source_selectable_example.tsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useMemo, useState } from 'react';
+import {
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+  EuiTitle,
+} from '@elastic/eui';
+import { MountPoint } from 'opensearch-dashboards/public';
+import { CoreStart } from 'opensearch-dashboards/public';
+import {
+  DataSourceManagementPluginSetup,
+  DataSourceSelectableConfig,
+} from 'src/plugins/data_source_management/public';
+import { ComponentProp } from './types';
+
+interface DataSourceSelectableExampleProps {
+  savedObjects: CoreStart['savedObjects'];
+  dataSourceEnabled: boolean;
+  notifications: CoreStart['notifications'];
+  setActionMenu?: (menuMount: MountPoint | undefined) => void;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+}
+
+export const DataSourceSelectableExample = ({
+  savedObjects,
+  dataSourceEnabled,
+  notifications,
+  setActionMenu,
+  dataSourceManagement,
+}: DataSourceSelectableExampleProps) => {
+  const DataSourceMenu = dataSourceManagement.ui.getDataSourceMenu<DataSourceSelectableConfig>();
+  const [selectedDataSources, setSelectedDataSources] = useState<string[]>([]);
+
+  const data: ComponentProp[] = [
+    {
+      name: 'savedObjects',
+      required: true,
+      defaultValue: '-',
+      description: 'The saved object client is used to fetch available data sources',
+      deprecated: false,
+    },
+    {
+      name: 'notifications',
+      required: true,
+      defaultValue: '-',
+      description: 'The notifications toasts object exposes interfaces to show toasts',
+      deprecated: false,
+    },
+    {
+      name: 'hideLocalCluster',
+      required: false,
+      defaultValue: 'false',
+      description: 'The property to hide local cluster from the data source selector',
+      deprecated: false,
+    },
+    {
+      name: 'fullWidth',
+      required: true,
+      defaultValue: '-',
+      description:
+        'The property of OutComboBox and when specified to true would expand to the entire width available',
+      deprecated: false,
+    },
+    {
+      name: 'displayAllCompatibleDataSources',
+      required: false,
+      defaultValue: 'undefined',
+      description: 'When specified to true, it will show all compatible data sources',
+      deprecated: false,
+    },
+    {
+      name: 'dataSourceFilter',
+      required: false,
+      defaultValue: 'undefined',
+      description:
+        'When specified, the filter function will be used to filter the available options before rendering',
+      deprecated: false,
+    },
+  ];
+
+  const columns: Array<EuiBasicTableColumn<ComponentProp>> = [
```

Comment:
Similar to the previous example, constant data appears to be copied across several files.

Reply:
Extracted columns as constant since it is definition of columns which are used across components